src/core/stream.c

```diff
@@ -567,7 +567,7 @@ QuicStreamShutdown(
     }
 
     if (!!(Flags & QUIC_STREAM_SHUTDOWN_FLAG_IMMEDIATE) &&
-        !Stream->Flags.ShutdownComplete) {
+        !Stream->Flags.ShutdownComplete && !Stream->Flags.RemoteCloseResetReliable) {
```

Comment:
Should we return an error instead of just ignoring the IMMEDIATE flag in this case?

Reply:
Just curious, why don't we throw an error if `Stream->Flags.ShutdownComplete = TRUE`?

Reply:
Sounds good, so if an app tries immediately shutting down after calling shutdown reliably, that's an error.

Reply:
MsQuic will return a transport error if the app tries immediately shutting down but it previously initiated or in the process of a reliable reset.